src/data/equiv/algebra.lean

```diff
@@ -328,3 +327,94 @@ instance symm.is_ring_hom {e : α ≃r β} : is_ring_hom e.to_equiv.symm := hom
   e.symm.to_equiv x = e.to_equiv.symm x := rfl
 
 end ring_equiv
+
+
+/-
+We define the type of automorphisms on groups, additive monoids,
+additive groups and rings using `mul_equiv`, `add_equiv` and `ring_equiv`.
+In each case this type also forms a group.
+-/
+
+@[extensionality] lemma mul_equiv.ext {α β : Type} [group α] [group β]
+{f g : mul_equiv α β} (h : f.to_fun = g.to_fun) : f = g :=
+by { cases f, cases g, congr, apply equiv.eq_of_to_fun_eq h }
+
+
+namespace group
+
+def aut (γ : Type) [group γ] := mul_equiv γ γ
+
+/--
+The group operation on automorphisms of a group is defined by
+λ g h, mul_equiv.trans h g.
+This means that multiplication agrees with composition, (g*h)(x) = g (h x) .
+-/
+instance aut_group (γ : Type) [group γ] : group (aut γ) :=
+{ mul := λ g h, mul_equiv.trans h g,
+  one := mul_equiv.refl γ,
+  inv := mul_equiv.symm,
+  mul_assoc := λ _ _ _, by { ext, refl },
+  one_mul := λ _, by { ext, refl },
+  mul_one := λ _, by { ext, refl },
+  mul_left_inv := λ _, by { ext, apply equiv.left_inv } }
+
+end group
+
+
+@[extensionality] lemma add_equiv.ext {α β : Type} [has_add α] [has_add β]
+{f g : add_equiv α β} (h : f.to_fun = g.to_fun) : f = g :=
+by { cases f, cases g, congr, apply equiv.eq_of_to_fun_eq h }
+
+namespace add_monoid
+
+def aut (α : Type) [has_add α] := add_equiv α α
```

Comment:
Why do you assume `has_add` over here, while you assume `group` in the multiplicative case?

Reply:
I've added the monoid case where I only assume ```has_mul```